uqcsbot/dominos_coupons.py

```diff
@@ -123,12 +182,14 @@ def keyword_matches(self, keyword: str) -> bool:
         return keyword.lower() in self.description.lower()
 
 
-def _get_coupons(n: int, ignore_expiry: bool, keywords: List[str]) -> List[Coupon]:
+def _get_coupons(
+    n: int, ignore_expiry: bool, keywords: List[str], source: str
+) -> List[Coupon]:
     """
     Returns a list of n Coupons
     """
 
-    coupons = _get_coupons_from_page()
+    coupons: List[Coupon] = _get_coupons_from_page(source)
```

Comment:
The explicit type on this line isn't needed. Feel free to leave it if you think it makes it more readable.